tools/work/README.md fixup

achadwick opened the pull request:
1. I fixed a few Markdown issues that were breaking some of my editors that have weaker Markdown processors.
    - Indented fenced code blocks are not widely supported, comments especially
    - The doc is still dependent on indented fences toward the end, but it degrades cleaner for single lines ;)

2. Added a list of exceptions where pure `#fff` and `#000` are allowed.

3. Updated the Debian commands for newer releases.
    - Scour is now `python3-scour`

5. Removed dangerous instruction to use `sudo pip install scour`
    - Mostly unnecessary, because Debian is now up to date, so don't give an example any more and demote the info
    - It was always dangerous to go messing with the Python libs tree like this
    - Debian has moved to externally-managed-environment, so pip works in fewer cases anyway
    - The modern advice is to use [pipx](https://github.com/pypa/pipx) on Debian 12 or later, particularly for comamnd line utils like scour

achadwick commented:
I'm still confused about what the docs mean by "compabillity with mostly GTK Themes", btw.

morganist commented:
this compatibility talks about the choise of standard black and white colours, an it appears that the present combinations were tested with different gtk colour themes. that's the reason why we even define them as standard and discourage the use of plain white and black colours.

morganist commented:
hey, great rewrite, I always wanted to at least improve the grammar and the wording, but never got to it. thanks! however, could you please change the section about exceptions that concerns the use of pure white? I believe it should be "it is appropriate to also use pure white" in  logos/text/other designs *on top of* colored icons, instead of an imperative. The base colours of Papirus can make pure white stand out too much if it is used in every situation, and almost all our present fullcolour icons use white elements in #e4e4e4

morganist commented:
it just that i generally believe #e4e4e4 should be used everywhere, and pure white should be used only if it looks really wrong.

achadwick commented:
@morganist 
Agreed. I'll make it clearer that I'm not talking about the base shapes. Better to say that pure '#ffffff' or '#000000' is _never_ appropriate for those base elements (apart from their highlights and shadows, with transparency). For logos or symbols atop the base, pure white or sometimes pure black can also be appropriate

(Aside: mediaeval heraldry would call the logos and symbols charges and the base shapes fields, so this is kinda like reinventing the [rule of tincture](https://en.wikipedia.org/wiki/Rule_of_tincture), I guess… well, not that pure black is a "metal". It might be a bit more modern to point at WCAG2.0 AA or AAA contrast ratio standards somewhere, though :sweat_smile:)

So, a plan for me in this update:

* Make the distinction between symbol and base clearer in the docs
* Make it clear that the "outline" handwave used elsewhere applies to both levels, or perhaps just say "clear shapes"
* Suggest choices that favour contrast and clarity of foreground to background
   - there are a few icons which have subpar choices for symbol:base (foreground:background) contrast already, so I think the docs could benefit from this update
   - it matters for accessibility especially. We could suggest a WCAG 2.0 level in a future edit. AAA would be very achievable, and AA would be better still, but tends to mute foreground elements on whiteish backgrounds
* Stress that Papirus's non-symbolic icons have to display well on both light and dark themes, potentially with pure white or black backgrounds themselves
   - increasingly so, actually, given direction of a lot of theme design these days

(Another aside: I'm actually more in favour of the Papirus "coloured base" style over the e4 grey or any other light grey or white as a paper colour. Why? I'm an obligate light theme user, and really often the tab of the common "document" background disappears in themes whose background aren't too glaring white, simply because the tabs are that bit lighter than the `e4` base shade: `fa` vs the theme's `f7`.

![example of what I'm talking about](https://github.com/user-attachments/assets/b4cf7f76-d319-478a-973c-190ce27d3389)

I don't want to rock the bot too much right now and start suggesting that nobody uses 
`#e4e4e4` grey bases for new icons though :wink: )

achadwick commented:
> it just that i generally believe #e4e4e4 should be used everywhere, and pure white should be used only if it looks really wrong.

Is that just for bases, or for the designs on top? If it's just for bases, I agree 100%. The designs though _very often_ benefit form the added contrast of using pure white and black, and for these overlaid designs it's actually established Papirus practice for the common design of a logo over a coloured rather than a monochrome background. Check out the mimetypes folder with an eye dropper tool ;)

achadwick commented:
### Is ff white too much?

> in #3921 for example, the logo is clearly too white for a Papirus icon,

I kinda disagree here even for a largish foreground element like the shield. I think the `e4` grey looks very dull indeed on a light background, and `ff` is just better to me, against the current Adwaita's *extremely* bright `#ff` active and `f7` inactive backgrounds, and also against the `ff` light mode here on Github. Adwaita is the current GNOME theme.

Comparing e4 ![example](https://github.com/user-attachments/assets/c5e2e8dc-7393-418b-8467-61a133cca411), ff ![example](https://github.com/user-attachments/assets/0038e11c-a967-438f-b5c6-f8e955ab2888), and e4 ![example](https://github.com/user-attachments/assets/fb075830-6ad2-4740-a8ac-6e4eded19fd4) as a potential compromise, only the ff and e4 look any good to me against modern Adwaita.

### Contrast

> and the contrast with the backgroung is sufficient to use #e4e4e4

All the e4/f4/f4 shades above are fine against the `#AA3FE4` example background at <https://webaim.org/resources/contrastchecker/> for graphical elements. However, only `#FFFFFF` is any good for ordinary size text (you never really get visually large text inside an icon unless it's a single letter!).

Arguably, there is a difference between sharp lines and big blocks for foreground elements. There, perhaps a good compromise would be to suggest an intermediate colour, like `#e4e4e4`.

Of course, that's just one example, and we don't yet tell contributors to check WCAG 2.0 contrast ration.

achadwick commented:
### Existing examples

There really are lots of examples in Papirus where pure white is used, even for very large elements, and the result is juicy and fresh and Papirus-ish. I think these are perfectly correct, and would look so much worse if e4 grey was used instead:

* in mimetypes:  ![image-x-generic](https://github.com/user-attachments/assets/45150b37-6b09-45a7-a917-38953990d084) ![audio-x-generic](https://github.com/user-attachments/assets/cc50c7cb-09da-4e2a-ba2b-678697dc550b) ![x-office-presentation](https://github.com/user-attachments/assets/5052c121-6cc3-4b63-97b7-b964695b0836) ![application-x-iso](https://github.com/user-attachments/assets/6635d7d0-2cc9-47d9-9130-b73833234191) ![text-vcard](https://github.com/user-attachments/assets/3070e217-831c-4d96-9faf-69e1f3dbf3a8) ![application-x-qemu-disk](https://github.com/user-attachments/assets/93a573a1-529c-4541-9a07-2a67b0640ef8) ![application-x-codeblocks-workspace](https://github.com/user-attachments/assets/790c55c7-7fa9-4bef-9264-476937664af4) ![application-x-designer](https://github.com/user-attachments/assets/7a5ecdf7-78ec-44da-9de1-da54e6235722)

* in apps: ![anjuta](https://github.com/user-attachments/assets/272466c0-182c-4b13-aa45-cc360d409b1e) ![AdobeFlash](https://github.com/user-attachments/assets/e5dddab3-19de-41cf-b0c7-9c7314a7a9b9) ![4kstogram](https://github.com/user-attachments/assets/b43ba708-69ab-4eeb-9583-ab891c979942) ![duolingo](https://github.com/user-attachments/assets/9b9144c4-78ed-4aac-9cbd-1502f8dd961e) ![gens-gs](https://github.com/user-attachments/assets/30a7c7bf-83ad-4d76-b604-872fd2e259a2) ![com github cassidyjames clairvoyant](https://github.com/user-attachments/assets/a5da9c0b-3f99-48df-9d8b-362034a59fce) ![clementine](https://github.com/user-attachments/assets/38da31ea-df63-43ff-9eac-98df7472050a)

Observations: I love that clementine icon so much. Mimetypes really are not special, I think. But notice how the examples have their ff element surrounded by darker colours or shades! That's what I'm trying to get across. The character icon eyes (in apps) would look especially weird if they were redone as e4. This is what I mean by a "foreground element" or "logo" element of the icon, and I think we need to make that distinction.

There are as many places where ff white **really should _not_ be used** for a "white" element, and is rightly avoided. Not really counterexample here: they're all examples where there are white or off-white objects that would sit directly on a potentially white background, and they should follow a different rule. I am not proposing to change the e4 ruling for these elements at all. This is what I mean by the "base" element of the icon.

* in mimetypes: ![text-css](https://github.com/user-attachments/assets/9ec0923e-5016-4d9a-997f-6e2d44f9b9ec) ![image-x-svg+xml](https://github.com/user-attachments/assets/eeac063c-4e1c-4000-91b4-edf74a1fbb2a) ![application-x-sqlite2](https://github.com/user-attachments/assets/30367152-c4bf-4626-882a-5b265f71a8fb) ![x-content-blank-cd](https://github.com/user-attachments/assets/f5f10571-c363-4225-84de-86d6df8f8409)

* in apps: ![ghostwriter](https://github.com/user-attachments/assets/bcc3ff97-da2b-455e-90be-d90e078b3864) ![com github tchx84 Flatseal](https://github.com/user-attachments/assets/a1b83458-ba63-4b50-bf1f-bd65e6102940) ![caffeine](https://github.com/user-attachments/assets/591bf7fe-bc69-4582-bd29-39b7cf08f3a4) ![accessories-text-editor](https://github.com/user-attachments/assets/0b577baa-66de-4898-b946-0d5e61baa2d3) ![nestopia](https://github.com/user-attachments/assets/073db709-4f10-4006-a535-b9804ce6707c)

achadwick commented:
In other words, I think the existing suggestion to use e4 grey and nothing brighter for white objects has arisen only from considering what I'm describing as a base element. It makes a lot less sense to apply it to foreground elements, and designers of established icons have ignored the e4 suggestion in apps and also in mimetypes for elements on top of the base, or surrounded by contrasting colours.

Anyway, I'll try and summarise all of that into this PR, and we'll see if that's acceptable.

achadwick commented:
> So do I change the white in #3921 or not? I'm OK either way. I would prefer something in between pure white and the grey, but also consistency is important.

@wilwarindi
I'd wait a little while, and leave it as it is till this is decided. @morganist has the commit rights, not me.

I don't think there's currently a fixed palette, and that's one of the changes I want to make in this README. I think it's better to think of the README's "black" and "white" as high and low brightness limits for base element's colour instead. They're like that to look as good as possible against different themes. Foreground elements are isolated from the theme background colour by the base elements, so they can go brighter or darker, as appropriate.

I'm a fan of that f4 light grey for an intermediate foreground colour, if e4 looks dull there and ff is too glaring. What do you think? I could suggest it in the doco revision.

achadwick commented:
I honestly think that ff white is perfectly appropriate in most situations where the base colour isn't too bright itself. the examples in https://github.com/PapirusDevelopmentTeam/papirus-icon-theme/pull/3912#issuecomment-2605924616 really do look good to me still.

achadwick commented:
OK, I hope this new draft makes everyone happy. I've really tried to clarify *why* those two black and white colours are stipulated. Better to think of them as brightness limits that make base shapes stand out against commonly available GTK themes.

Surprising Papirus fact: pure black really is only used for shadows. Its usage is totally different from pure white, which is *a lot* more widely used - but only really for foreground designs.

:point_right:  https://github.com/achadwick/papirus-icon-theme/blob/toolswork-readme-fixup/tools/work/README.md 

It's getting a bit big now, and I could split it.

morganist commented:
@achadwick you opened my eyes to the fact that brighter white is really widely used, as it turns out. I never really noticed any clash between similar icons with different shades of white on the foreground when they are placed on a dock or a panel, so I guess you are totally correct about the use of white colours.
I reviewed your improved readme and I'll say it's ready to merge!

achadwick commented:
I think that's it, ready for proofreading! A few late changes just now:

* I split out the design notes into a separate document to get it away from all the notes about the scripted workflows.
* Added a note to DESIGN.md about how to avoid using gradients.
* Various typos and grammatical fixes.
* I've updated the main README to point at the new docs, to aid onboarding.

Over to you, @morganist :D

morganist commented:
that's an awesome change to our documentation, thank you very much!
